src/main/java/Duke.java

```diff
@@ -1,10 +1,145 @@
+import java.util.Scanner;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class Duke {
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        int counter = 0;
+        ArrayList<Task> list = new ArrayList<>();
+
+        File tasks = new File("./data/duke.txt");
+        try {
+            if (tasks.exists()) {
+                Scanner sc1 = new Scanner(tasks);
+                String data;
+                while (sc1.hasNextLine()) {
+                    data = sc1.nextLine();
+                    String[] dataSplit = data.split(" \\| ");
+                    switch (dataSplit[0]) {
+                        case "T":
+                            list.add(new ToDos(Boolean.parseBoolean(dataSplit[1]), dataSplit[2]));
+                            break;
+                        case "D":
+                            list.add(new Deadlines(Boolean.parseBoolean(dataSplit[1]), dataSplit[2], dataSplit[3]));
+                            break;
+                        case "E":
+                            list.add(new Events(Boolean.parseBoolean(dataSplit[1]), dataSplit[2], dataSplit[3]));
+                    }
+                    counter++;
+                }
+                sc1.close();
+            } else {
+                File directory = new File("./data");
+                if (!directory.exists()) {
+                    directory.mkdir();
+                }
+                tasks.createNewFile();
+            }
+        } catch (IOException e) {
+            System.out.println("An error has occurred while reading duke.txt");
+        }
+
+        Scanner sc = new Scanner (System.in);
+        System.out.println("Hello! What can I do for you:>");
+        String input = sc.nextLine();
+        while (!input.equals("bye")) {
+            if (input.equals("list")) {
+                if (list.size() == 0) {
+                    System.out.println("You have no tasks in the list!");
+                }
+                for (int i = 0; i < list.size(); i++) {
+                    System.out.println((i + 1) + "." + list.get(i));
+                }
+            } else {
+                String[] inputSplit = input.split(" ");
+                switch (inputSplit[0]){
+                    case "done":
+                        if (inputSplit.length < 2){
```

Comment:
Very minor but remember to leave white space before "{" just for consistency and readability 😄